src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfGeneFeature.java

```diff
@@ -0,0 +1,121 @@
+                                  AnnotationSource annotationSource,
+                                  FeatureType featureType,
+                                  int genomicStartLocation,
+                                  int genomicEndLocation,
+                                  GenomicStrand genomicStrand,
+                                  GenomicPhase genomicPhase,
+                                  String geneId,
+                                  String transcriptId,
+                                  GeneTranscriptType geneType,
+                                  GeneTranscriptStatus geneStatus,
+                                  String geneName,
+                                  GeneTranscriptType transcriptType,
+                                  GeneTranscriptStatus transcriptStatus,
+                                  String transcriptName,
+                                  int exonNumber,
+                                  String exonId,
+                                  LocusLevel locusLevel,
+                                  ArrayList<OptionalField<?>> optionalFields,
+                                  String anonymousOptionalFields) {
+
+        super(lineNumber, chromosomeName, annotationSource, featureType, genomicStartLocation, genomicEndLocation, genomicStrand, genomicPhase, geneId, transcriptId, geneType, geneStatus, geneName, transcriptType, transcriptStatus, transcriptName, exonNumber, exonId, locusLevel, optionalFields, anonymousOptionalFields);
+    }
+
+    public static GencodeGtfFeature create(long lineNumber,
+                                           String chromosomeName,
+                                           AnnotationSource annotationSource,
+                                           FeatureType featureType,
+                                           int genomicStartLocation,
+                                           int genomicEndLocation,
+                                           GenomicStrand genomicStrand,
+                                           GenomicPhase genomicPhase,
+                                           String geneId,
+                                           String transcriptId,
+                                           GeneTranscriptType geneType,
+                                           GeneTranscriptStatus geneStatus,
+                                           String geneName,
+                                           GeneTranscriptType transcriptType,
+                                           GeneTranscriptStatus transcriptStatus,
+                                           String transcriptName,
+                                           int exonNumber,
+                                           String exonId,
+                                           LocusLevel locusLevel,
+                                           ArrayList<OptionalField<?>> optionalFields,
+                                           String anonymousOptionalFields) {
```

Comment:
My earlier comment in the base class about introducing a struct class to hold these values also applies to the various subclasses.